src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/PKCS1MaskGenerationMethod.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public class PKCS1MaskGenerationMethod : MaskGenerationMethod
+    {
+        private string _hashNameValue;
+        
+        public PKCS1MaskGenerationMethod()
+        {
+            _hashNameValue = "SHA1";
+        }
+
+        public string HashName
+        {
+            get { return _hashNameValue; }
+            set 
+            { 
+                if (value != null && value != "SHA1")
+                {
+                    throw new PlatformNotSupportedException();
+                }
+
+                _hashNameValue = "SHA1";
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA5350", Justification = "CryptoConfig.CreateFromName is not supported on Unix platforms, should create SHA1 directly")]
+        public override byte[] GenerateMask(byte[] rgbSeed, int cbReturn)
+        {
+            HashAlgorithm hash = SHA1.Create();
+            byte[] rgbCounter = new byte[4];
+            byte[] rgbT = new byte[cbReturn];
+
+            uint counter = 0;
+            for (int ib = 0; ib < rgbT.Length;)
+            {
+                //  Increment counter -- up to 2^32 * sizeof(Hash)
+                Helpers.ConvertIntToByteArray(counter++, ref rgbCounter);
+                hash.TransformBlock(rgbSeed, 0, rgbSeed.Length, rgbSeed, 0);
+                hash.TransformFinalBlock(rgbCounter, 0, 4);
+                byte[] _hash = hash.Hash;
```

Comment:
Naming: Since the hash is a local it should be `hash`, not `_hash` (the proper name for an instance field).  To disambiguate it from the HashAlgorithm object, I'd rename the HashAlgorithm object to `hasher` (the thing which hashes)